pkg/azure/access/helpers/vm.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2023 SAP SE or an SAP affiliate company
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helpers
+
+import (
+	"context"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v5"
+	"k8s.io/klog/v2"
+
+	"github.com/gardener/machine-controller-manager-provider-azure/pkg/azure/access/errors"
+	"github.com/gardener/machine-controller-manager-provider-azure/pkg/azure/instrument"
+	"github.com/gardener/machine-controller-manager-provider-azure/pkg/azure/utils"
+)
+
+// labels used for recording prometheus metrics
+const (
+	vmGetServiceLabel    = "virtual_machine_get"
+	vmUpdateServiceLabel = "virtual_machine_update"
+	vmDeleteServiceLabel = "virtual_machine_delete"
+	vmCreateServiceLabel = "virtual_machine_create"
+)
+
+// Default timeouts for all async operations - Create/Delete/Update
+const (
+	defaultDeleteVMTimeout = 15 * time.Minute
+	defaultCreateVMTimeout = 15 * time.Minute
+	defaultUpdateVMTimeout = 10 * time.Minute
```

Comment:
could you again tell why `defaultUpdateVMTimeout` != `defaultDeleteVMTimeout` ? Sorry we forgot.

Reply:
updating VM description is cheaper than the creation/deletion so this makes sense, though exact values can be debated.

Reply:
We can change this to 15 mins if you want, i am open to it.

Reply:
> updating VM description (setting cascade delete for nics/disks) is cheaper than the creation/deletion so this makes sense, though exact values can be debated.

Okay. Could you add a comment specifying this.

Reply:
I have added a comment now.